video/out/vo_dmabuf_wayland.c

```diff
@@ -0,0 +1,357 @@
+
+        entry->drm_format = layer.format;
+        for (plane_no = 0; plane_no < layer.nb_planes; ++plane_no) {
+            AVDRMPlaneDescriptor plane = layer.planes[plane_no];
+            int object_index = plane.object_index;
+            AVDRMObjectDescriptor object = avdesc->objects[object_index];
+            uint64_t modifier = object.format_modifier;
+
+            zwp_linux_buffer_params_v1_add(params, object.fd, plane_no,
+                                                   plane.offset, plane.pitch, modifier >> 32, modifier & 0xffffffff);
+        }
+    }
+
+    return true;
+}
+#endif
+
+static void resize(struct vo *vo)
+{
+    struct vo_wayland_state *wl = vo->wl;
+    struct mp_rect src;
+    struct mp_rect dst;
+    struct mp_osd_res osd;
+    const int width = wl->scaling * mp_rect_w(wl->geometry);
+    const int height = wl->scaling * mp_rect_h(wl->geometry);
+    
+    vo_wayland_set_opaque_region(wl, 0);
+    vo->dwidth = width;
+    vo->dheight = height;
+    vo_get_src_dst_rects(vo, &src, &dst, &osd);
+
+    if (wl->viewport)
+        wp_viewport_set_destination(wl->viewport, 2 * dst.x0 + mp_rect_w(dst), 2 * dst.y0 + mp_rect_h(dst));
+
+    if (wl->video_viewport)
+        wp_viewport_set_destination(wl->video_viewport, mp_rect_w(dst), mp_rect_h(dst));
+    wl_subsurface_set_position(wl->video_subsurface, dst.x0, dst.y0);
+    vo->want_redraw = true;
+}
+
+static void draw_frame(struct vo *vo, struct vo_frame *frame)
+{
+    struct priv *p = vo->priv;
+    struct vo_wayland_state *wl = vo->wl;
+    struct wlbuf_pool_entry *entry;
+    
+    if (!vo_wayland_check_visible(vo))
+        return;
+
+    // ensure the pool is reset after hwdec seek,
+    // to avoid stutter artifact
+    p->reset_count++;
+    if (p->want_reset &&  p->reset_count <= 2){
+        wlbuf_pool_clean(p->wlbuf_pool);
+        if (p->reset_count == 2)
+            p->want_reset = false;
+    }
+
+    /* lazy initialization of buffer pool */
+    if (!p->wlbuf_pool) {
+#if HAVE_VAAPI
+        p->display = (VADisplay)ra_get_native_resource(p->ctx->ra, "VADisplay");
+        if (p->display)
+            p->wlbuf_pool = wlbuf_pool_alloc(vo, wl, vaapi_key_provider, vaapi_dmabuf_importer);
+#endif
+#if HAVE_DRM
+        if (!p->wlbuf_pool)
+            p->wlbuf_pool = wlbuf_pool_alloc(vo, wl, drmprime_key_provider, drmprime_dmabuf_importer);
+#endif
+    }
+    entry = wlbuf_pool_get_entry(p->wlbuf_pool, frame->current);
+    if (!entry)
+        return;
+
+    MP_VERBOSE(entry->vo, "Schedule buffer pool entry : %lu\n",entry->key );
+    wl_surface_attach(wl->video_surface, entry->buffer, 0, 0);
+    wl_surface_damage_buffer(wl->video_surface, 0, 0, INT32_MAX, INT32_MAX);
+}
+
+static void flip_page(struct vo *vo)
+{
+    struct vo_wayland_state *wl = vo->wl;
+
+    wl_surface_commit(wl->video_surface);
+    wl_surface_commit(wl->surface);
+    if (!wl->opts->disable_vsync)
+        vo_wayland_wait_frame(wl);
+    if (wl->present)
```

Comment:
It does actually compile but I meant the following hah.

```suggestion
    if (wl->use_present)
```